modules/generic-scala3/src/main/scala/pureconfig/generic/scala3/AnyValDerivationMacros.scala

```diff
@@ -0,0 +1,72 @@
+package pureconfig
+package generic
+package scala3
+
+import scala.compiletime._
+import scala.deriving.Mirror
+import scala.quoted._
+
+private[pureconfig] object AnyValDerivationMacros {
+
+  /** Derive a `ConfigReader` for a value class. Can only be called after checking that `A` is a value class.
+    */
+  inline def unsafeDeriveAnyValReader[A]: ConfigReader[A] = ${ deriveAnyValReaderImpl[A] }
+
+  def deriveAnyValReaderImpl[A: Type](using Quotes): Expr[ConfigReader[A]] = {
```

Comment:
Can we make all these private?

```suggestion
  private def deriveAnyValReaderImpl[A: Type](using Quotes): Expr[ConfigReader[A]] = {
```